dts/bindings/pinctrl/microchip,xec-pinctrl.yaml

```diff
@@ -0,0 +1,126 @@
+    node to select peripheral pin functions.
+
+    The node has the 'pinctrl' node label set in your SoC's devicetree,
+    so you can modify it like this:
+
+      &pinctrl {
+              /* your modifications go here */
+      };
+
+    All device pin configurations should be placed in child nodes of the
+    'pinctrl' node, as shown in the spi0 example below:
+
+    A group can also specify shared pin properties common to all the specified
+    pins, such as the 'bias-pull-up' property in group 2. Here is a list of
+    supported standard pin properties:
+
+    - bias-disable: Disable pull-up/down (default behavior, not required).
+    - bias-pull-down: Enable pull-down resistor.
+    - bias-pull-up: Enable pull-up resistor.
+    - drive-push-pull: Output driver is push-pull (default, not required).
+    - drive-open-drain: Output driver is open-drain.
+    - output-high: Set output state high when pin configured.
+    - output-low: Set output state low when pin configured.
+
+    Custom pin properties for drive strength and slew rate are available:
+    - drive-strength
+    - slew-rate
+
+    Driver strength and slew rate hardware defaults vary by SoC and pin.
+    The "hw-default" setting tell the pinctrl code to not touch the slew
+    rate or drive strength settings for the respective pin.
+
+    An example for MEC172x family, include the chip level pinctrl
+    DTSI file in the board level DTS:
+
+      #include <microchip/mec172x/mec172xnsz-pinctrl.dtsi>
+
+      We want to use the shared SPI port of the MEC172x QMSPI controller
+      and want the chip select 0 to be open-drain.
+
+      To change a pin's pinctrl default properties add a reference to the
+      pin in the board's DTS file and set the properties.
+
+      &spi0 {
+        pinctrl-0 = < &shd_cs0_n_gpio055
+                      &shd_clk_gpio056
+                      &shd_io0_gpio223
+                      &shd_io1_gpio224
+                      &shd_io3_gpio016 >;
+        pinctrl-names = "default";
+     }
+
+     &shd_cs0_n_gpio055 {
+        drive-open-drain;
+     };
+
+compatible: "microchip,xec-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node.yaml
+      child-binding:
+        property-allowlist:
+          - bias-disable
+          - bias-pull-down
+          - bias-pull-up
+          - drive-push-pull
+          - drive-open-drain
+          - low-power-enable
+          - output-high
+          - output-low
+
+properties:
+    reg:
+      required: true
+
+child-binding:
+    description: |
+      This binding gives a base representation of the Microchip XEC pins
+      configration
+
+    properties:
+        pinmux:
+            type: int
+            required: true
+            description: Pinmux selection
+
+        gpioph:
+            type: phandle
+            required: true
+            description: phandle to GPIO node
+
+        slew-rate:
+          required: false
+          type: string
+          default: "low-speed"
```

Comment:
All use of defaults must be justified. Please read the rules in https://docs.zephyrproject.org/latest/guides/dts/bindings.html#rules-for-mainline-bindings and update this PR accordingly

Reply:
Will add default value in the description

Reply:
Updated description with default value.

Reply:
Sorry, but that wasn't my point. I am not asking for the default to be placed into the description. I am asking for the default to be _justified_, according to the rules in my above link. Please revisit.

Reply:
The pinctrl driver will not program the slew-rate (or the drive-strength) if the those properties are not defined in the dts. As such we are considering to remove the default field (for slew-rate and drive-strength) in the yaml file.
Can you confirm?

Note: we don't want to program the slew-rate (or drive-strength) for each pin used; we want to use the hardware defaults most of the time.

Reply:
A question here: if "hardware defaults" mean no-op, how do I know that pin is correctly configured? What if e.g. a bootloader configures pins with a different set of options?

Reply:
I have the same question, but I want to clarify something else.

It's OK to use `default` in the binding, if and only if you also justify **why** the default was chosen in the description, **and** that justification follows the rules in my link. So keeping the default is OK -- especially to cover cases like the bootloader situation in @gmarull 's question -- you just have to justify it.

Just writing "the default value of X is the SoC power-on-reset value" is fine as long as the value is independently configurable from other values  as described in the bindings documentation.

Is that clear?

Reply:
Okay, we will keep default in the binding and add "The default value of slew-rate is the SoC power-on-reset value" in the description.
Yes, this value is independently configurable from other values.

Reply:
> A question here: if "hardware defaults" mean no-op, how do I know that pin is correctly configured? What if e.g. a bootloader configures pins with a different set of options?

The pins will be in the hardware default state (as in the datasheet) except for those configured by the boot rom loader. We are working on how to make available this information (boot rom configured pins). 

Can we share the boot rom configured pins table in [CPGZephyrDocs/MEC172x]([CPGZephyrDocs](https://github.com/MicrochipTech/CPGZephyrDocs)/MEC172x/) and update the board's [rst](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/mec172xevb_assy6906/doc/index.rst) file with its link?
If so, I will be creating a separate PR for updating the .rst.

Reply:
> Okay, we will keep default in the binding and add "The default value of slew-rate is the SoC power-on-reset value" in the description.
> Yes, this value is independently configurable from other values.

Sounds good, thanks. I see you're making a lot of contributions lately, which is great! Please remember the rules for upstream bindings in my link and apply them consistently.